### DISCUSSION TITLE
Mismatch on entity names ERROR

### DISCUSSION THREAD

**Discussion by ITSpecialist111** [nathanmarlor/foxess_modbus_charge_period_card#8] [Support]
I got an issue in home assistant. There the lovelace card that has an error:

Unable to load the following entities
sensor.foxess_modbus._time_period_1_start,
sensor.foxess _modbus time period 1 end
binary_sensor.foxess_modbus_time_period_1_enable_for
binary_sensor.foxess_modbus_time_period _1_enable_ch
sensor.foxess_modbusltimelperiod_2.start

But then I check the entity for the FOX Modbus and all the sensor names don't include the words foxess_modbus. 
   E.g.;      sensor.time_period_1_end

**Comment by canton7** [comment]
Weird. Can you post the contents of the file `/config/.storage/core.config_entries`?

**Reply by ITSpecialist111** [reply]
I think I realise the error. I've got both the Cloud and Modbus integrations enabled. So the entities can't be the same name. I'm trying to unpick at the moment and reinstall

**Reply by canton7** [reply]
The charge period card asks the foxess_modbus integration what the entity ids of the charge period entities are, so if they're wrong then foxess_modbus is very confused

**Reply by canton7** [reply]
Although, if you've gone into home assistant and manually changed the entity ids, we can't cope with that

**Reply by ITSpecialist111** [reply]
No not changed manually. The Cloud version was installed 1st but that has a prefix of:  sensor.foxess_bat_discharge.  But the modbus ones don't have any fox prefix: e.g.  Battery Discharge (FoxEss ModBus Unit)
sensor.battery_discharge

**Reply by ITSpecialist111** [reply]
![Screenshot 2023-08-24 215324](https://github.com/nathanmarlor/foxess_modbus_charge_period_card/assets/142685548/3b161d7a-62a5-46b5-bd95-e5983c0982ab)

**Reply by canton7** [reply]
OK. Back to my first question then I guess - can you post the contents of that file?

**Reply by ITSpecialist111** [reply]
just trying to get now

**Reply by ITSpecialist111** [reply]
Am struggling to get the samba service running atm :/ any other way?

**Reply by canton7** [reply]
How did you install home assistant?

**Reply by ITSpecialist111** [reply]
It's running in HyperV

**Reply by canton7** [reply]
Can you access it's fikeaystem in any way? If it's home assistant operating system, I think you can get an editor in the we UI? Not done it myself

**Reply by ITSpecialist111** [reply]
you can't see hidden/protected folders using the file editor in the UI. trying another way

**Reply by ITSpecialist111** [reply]
Hey, looks like I needed to sleep on it. Here's the file
~~core.config_entries.txt~~

**Reply by canton7** [reply]
```
"entity_id_prefix": "foxess_modbus_unit"
```

So foxess_modbus will be asking home assistent to prefix all entity ids with that value. I'm not sure why home assistent isn't doing that... You'll probably find that the Reinmann integral sensors (such as pv1_energy_total) aren't working either?

The thing to do is probably remove the entity_id_prefix. You can just edit that file, or remove / re-add the integration - either leave the "friendly name" box blank, or if you do enter something, check the box to manually specify the entity id prefix then leave that blank.

**Reply by ITSpecialist111** [reply]
Thank you for looking at it! I did try to remove the integration and read but was the same effect. I'll try adjusting the file and go from there 👯

**Reply by ITSpecialist111** [reply]
PV energy total just says Unknown- you're correct

**Comment by canton7** [comment]
Looks like the same as #9, which I've just fixed in v1.7.3 of foxess_modbus.
